src/Microsoft.DotNet.Cli.Utils/CommandResolution/MSBuildProject.cs

```diff
@@ -124,7 +124,9 @@ public LockFile GetLockFile()
             var lockFilePath = GetLockFilePathFromProjectLockFileProperty() ??
                 GetLockFilePathFromIntermediateBaseOutputPath();
 
-            return new LockFileFormat().Read(lockFilePath);
+            return new LockFileFormat()
+                .ReadWithLock(lockFilePath)
```

Comment:
Lock in this context can mean both ProjectLock file and ConcurrencyLock (even though ReadWithLock suggests the latter it is quite confusing on the first glance), I think a rename would make sense